include/oneapi/tbb/detail/_flow_graph_join_impl.h

```diff
@@ -1026,6 +1026,7 @@
                            public sender<OutputTuple> {
     protected:
         using graph_node::my_graph;
+        using input_type = typename join_node_FE<JP, InputTuple, OutputTuple>::input_type;
```

Comment:
I have maid a quick check and it seems like this alias it is not needed outside the `join_node_FE` class - as well as other public aliases defined there. It seems like it would be better to make `input_type` and other aliases in `join_node_FE` specializations private. @aleksei-fedotov, what do you think?

Reply:
For what it`s worth. a public class like `join_node` definitely should hide all internal interfaces inherited from the implementation.

Reply:
> I have maid a quick check and it seems like this alias it is not needed outside the `join_node_FE` class - as well as other public aliases defined there. It seems like it would be better to make `input_type` and other aliases in `join_node_FE` specializations private. @aleksei-fedotov, what do you think?

Sorry for a late reply. 
The idea behind input and output type is to make node interfaces common so that users can reference the corresponding type easily by writing something like: `<node_type>::input_type` or `<node_type>::output_type`.
There are exceptions for multi-input/output node types (such as `join_node`, `split_node`). 
Yes, internal interfaces should be hidden. So, the changes are correct (except for the line with static assert).